test/cleanuphook.cc

```diff
@@ -0,0 +1,31 @@
+#include "napi.h"
+
+using namespace Napi;
+
+#if (NAPI_VERSION > 2)
+
+class TestCleanupHook : public CleanupHook {
+  public:
+    static void DoCleanup(const CallbackInfo& info) {
+	  static int secret = 42;
+	  static int wrongSecret = 17;
+	  TestCleanupHook* testCleanupHook = new TestCleanupHook(info, wrongSecret);
+	  delete testCleanupHook;
```

Comment:
If you want to delete the object here, do we really need to create the object on heap?